pandas_schema/schema.py

```diff
@@ -38,14 +42,28 @@ def validate(self, df: pd.DataFrame) -> typing.List[ValidationWarning]:
         """
         errors = []
 
-        # It's an error if the number of columns in the schema and data frame are different
         df_cols = len(df.columns)
-        schema_cols = len(self.columns)
-        if df_cols != schema_cols:
-            errors.append(ValidationWarning(
-                'Invalid number of columns. The schema specifies {}, but the data frame has {}'.format(schema_cols,
+
+        # If no columns are passed, validate against every column in the schema
+        # This is the default behaviour
+        if columns == []:
```

Comment:
columns should default to `None`, and you should check `is None` here instead. Why allocate an empty array when it won't be used for anything?